### PR TITLE
config: parse config as JSON instead of YAML

### PR DESCRIPTION
**Pull request by crawford**
This needs some more tests, but I want to get some feedback.

### CODE REVIEW

**Comment by marineam** [comment]
woo json++ :) </peanutgallery>

File: config/filesystem.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	nd := DevicePath(td)
+	if err := nd.assertValid(); err != nil {
+		return err
+	}
+	*d = nd
+	return nil
+}
+
+func (d DevicePath) assertValid() error {
+	if !strings.HasPrefix(string(d), "/dev") {
```
Comment (vcaputo) [inline config/filesystem.go]: Formatting filesystems in files (e.g. for loopback use) can occur outside of /dev, should someone want to do that with ignition why prohibit them?

File: config/filesystem.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+func (f FilesystemFormat) assertValid() error {
+	switch f {
+	case "ext4", "btrfs":
+		return nil
+	default:
+		return errors.New("invalid filesystem")
+	}
+}
+
+type MkfsOptions string
```
Comment (vcaputo) [inline config/filesystem.go]: []string as discussed

**Comment by vcaputo** [comment]
lgtm!

**Comment by crawford** [comment]
@vcaputo I added some tests.

**Comment by vcaputo** [comment]
hooray

**Pull request merged**